[ONNX] Partially re-enable RoiAlign and RoiPool unit tests

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #86216
* __->__ #86169
* #86168

This PR depends on https://github.com/pytorch/vision/pull/6685

BowenBao committed:
[ONNX] Partially re-enable RoiAlign and RoiPool unit tests

[ghstack-poisoned]

justinchuby commented on test/onnx/pytorch_test_common.py:
```diff
@@ -107,13 +107,15 @@ def wrapper(self, *args, **kwargs):
 
 
-def skipScriptTest(min_opset_version=float("inf")):
+def skipScriptTest(min_opset_version=float("inf"), reason=""):
```
Love this

justinchuby replied:
Can `min_opset_version=float("inf")` become `-1`? Feels like `float("inf")` is a little extreme

BowenBao replied:
Clarified in #86216

justinchuby commented on test/onnx/pytorch_test_common.py:
```diff
@@ -107,13 +107,15 @@ def wrapper(self, *args, **kwargs):
             if not self.is_script_test_enabled and self.is_script:
-                raise unittest.SkipTest("Skip verify test for TorchScript")
+                raise unittest.SkipTest(
+                    ": ".join(["Skip verify test for TorchScript", reason])
```
Use an fstring?

BowenBao replied:
didn't like the extra logic of checking empty str otherwise introducing ": " for nothing, any cleaner solution?

justinchuby replied:
I see what is going on here. 🤔 ` "Skip verify test for TorchScript" + (f": {reason}" if reason else "")` but not exactly elegant

justinchuby replied:
Or: Default to "no reason provided" to encourage ppl to add a reason?

BowenBao replied:
I'm thinking breaking it by "." So we won't complicate ourselves :)

BowenBao commented on test/onnx/pytorch_test_common.py:
```diff
@@ -107,13 +107,15 @@ def wrapper(self, *args, **kwargs):
             if not self.is_script_test_enabled and self.is_script:
-                raise unittest.SkipTest("Skip verify test for TorchScript")
+                raise unittest.SkipTest(
+                    ": ".join(["Skip verify test for TorchScript", reason])
```
```suggestion
```

BowenBao commented on test/onnx/pytorch_test_common.py:
```diff
@@ -107,13 +107,15 @@ def wrapper(self, *args, **kwargs):
-                raise unittest.SkipTest("Skip verify test for TorchScript")
+                raise unittest.SkipTest(
+                    ": ".join(["Skip verify test for TorchScript", reason])
+                )
```
```suggestion
```

BowenBao commented on test/onnx/pytorch_test_common.py:
```diff
@@ -107,13 +107,15 @@ def wrapper(self, *args, **kwargs):
             self.is_script_test_enabled = self.opset_version >= min_opset_version
             if not self.is_script_test_enabled and self.is_script:
-                raise unittest.SkipTest("Skip verify test for TorchScript")
+                raise unittest.SkipTest(
```
```suggestion
                raise unittest.SkipTest(f"Skip verify test for TorchScript. {reason}")
```

BowenBao committed:
Apply suggestions from code review

BowenBao committed:
Update on "[ONNX] Partially re-enable RoiAlign and RoiPool unit tests"


This PR depends on https://github.com/pytorch/vision/pull/6685


[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Partially re-enable RoiAlign and RoiPool unit tests"


This PR depends on https://github.com/pytorch/vision/pull/6685


[ghstack-poisoned]

AllenTiTaiWang commented on torch/_C/__init__.pyi.in:
```diff
@@ -602,7 +602,9 @@ class Node:
     def ts_(self, name: str, val: List[Tensor]) -> Node: ...
+    def ty(self, name: str) -> JitType: ...
     def ty_(self, name: str, val: JitType) -> Node: ...
+    def tys(self, name: str) -> List[JitType]: ...
```
Curious about the reason of adding these two here

BowenBao replied:
These were missing. It was discovered due to that the torch script version of `RoiAlign` and `RoiPool` used it. Although later it turned out we could not support the torch script version due to `prim::isinstance`.

BowenBao committed:
Update on "[ONNX] Partially re-enable RoiAlign and RoiPool unit tests"


This PR depends on https://github.com/pytorch/vision/pull/6685


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge -g

BowenBao commented on torch/_C/__init__.pyi.in:
```diff
@@ -603,7 +603,9 @@ class Node:
     def t_(self, name: str, val: Tensor) -> Node: ...
     def ts(self, name: str) -> List[Tensor]: ...
     def ts_(self, name: str, val: List[Tensor]) -> Node: ...
+    def ty(self, name: str) -> JitType: ...
```
```suggestion
```

BowenBao commented on torch/_C/__init__.pyi.in:
```diff
@@ -603,7 +603,9 @@ class Node:
     def ts_(self, name: str, val: List[Tensor]) -> Node: ...
+    def ty(self, name: str) -> JitType: ...
     def ty_(self, name: str, val: JitType) -> Node: ...
+    def tys(self, name: str) -> List[JitType]: ...
```
```suggestion
```

BowenBao committed:
Remove unneeded changes

BowenBao commented on torch/csrc/jit/python/python_ir.cpp:
```diff
@@ -767,11 +767,17 @@ void initPythonIRBindings(PyObject* module_) {
           [](Node& n, const char* name, const TypePtr& type) {
             return n.ty_(Symbol::attr(name), type);
           })
+      .def(
```
```suggestion
```

BowenBao committed:
cont

BowenBao commented:
@pytorchbot merge -g

abock commented:
@pytorchbot merge -g
